crates/ark/src/methods.rs

```diff
@@ -0,0 +1,154 @@
+            call.param(name.as_str(), value);
+        }
+
+        let result = call.call_in(ARK_ENVS.positron_ns)?;
+
+        // No method for that object
+        if result.sexp == r_null() {
+            return Ok(None);
+        }
+
+        // Convert the result to the expected return type
+        match result.try_into() {
+            Ok(value) => Ok(Some(value)),
+            Err(err) => Err(anyhow!("Conversion failed: {err:?}")),
+        }
+    }
+
+    pub fn register_method(&self, class: &str, method: RObject) -> anyhow::Result<()> {
+        let generic_name: &str = self.into();
+        RFunction::new("", ".ark.register_ark_method")
+            .add(RObject::try_from(generic_name)?)
+            .add(RObject::try_from(class)?)
+            .add(method)
+            .call_in(ARK_ENVS.positron_ns)?;
+        Ok(())
+    }
+
+    pub fn register_method_from_package(&self, class: &str, package: &str) -> anyhow::Result<()> {
+        // We store a call instead of the function itself to make sure:
+        // 1. Everything worked seamlessly with devtools::load_all() and similar functions (no risk of calling an
+        //    outdated method from a stale cache);
+        // 2. An escape hatch remained available for monkey-patching a method in the package namespace
+        //    (e.g., to enable workarounds for misbehaving methods).
+        let method = RCall::new(unsafe { r_symbol!(":::") })
+            .add(RObject::from(package))
+            .add(RObject::from(format!("{self}.{class}")))
+            .build();
+        self.register_method(class, method)?;
+        Ok(())
+    }
+
+    // Checks if a symbol name is a method and returns it's class
+    fn parse_method(name: &String) -> Option<(Self, String)> {
+        for method in ArkGenerics::iter() {
+            let method_str: &str = method.clone().into();
+            if name.starts_with::<&str>(method_str) {
+                if let Some((_, class)) = name.split_once(".") {
+                    return Some((method, class.to_string()));
+                }
+            }
+        }
+        None
+    }
+}
+
+pub fn populate_methods_from_loaded_namespaces() -> anyhow::Result<()> {
+    let loaded = RFunction::new("base", "loadedNamespaces").call()?;
+    let loaded: Vec<String> = loaded.try_into()?;
+
+    for pkg in loaded.into_iter() {
+        populate_methods_table(pkg.as_str()).or_log_error("Failed populating methods");
+    }
+
+    Ok(())
+}
+
+pub fn populate_methods_table(package: &str) -> anyhow::Result<()> {
+    let ns = r_ns_env(package)?;
+    let symbol_names = ns
+        .iter()
+        .filter_map(Result::ok)
+        .filter(|b| match b.value {
+            BindingValue::Standard { .. } => true,
+            BindingValue::Promise { .. } => true,
+            _ => false,
+        })
+        .map(|b| -> String { b.name.into() });
+
+    for name in symbol_names {
+        if let Some((generic, class)) = ArkGenerics::parse_method(&name) {
```

Comment:
It feels kind of inefficient to me to have to generate the vector of `String`s for each `ArkGeneric` each time you go through `parse_method()`. It's like we want to generate the vector of generic names once up front and reuse them for all symbol names we compare against.

Reply:
Removed automatic registration in https://github.com/posit-dev/ark/pull/560/commits/5ab5b9dcd0bca352bd0db6c369bf8eae33896d88